File: python/src/wdl_model/model/errors/wdl_import_exception.py
"""Exception raised when an import URI cannot be resolved or loaded."""

from __future__ import annotations

from .wdl_exception import WdlException
from .wdl_import_error import WdlImportError


class WdlImportException(WdlException):
    """Exception raised when an import URI cannot be resolved or loaded."""

    def __init__(
        self, message: str, import_location: str, cause: Exception | None = None
    ):
        super().__init__([WdlImportError(message, import_location)])
        if cause is not None:
            self.__cause__ = cause

File: typescript/src/model/errors/wdl-import-exception.ts
import { WdlException } from './wdl-exception.js';
import { WdlImportError } from './wdl-import-error.js';

/** Exception raised when an import URI cannot be resolved or loaded. */
export class WdlImportException extends WdlException {
  public constructor(message: string, importLocation: string, cause?: unknown) {
    super([new WdlImportError(message, importLocation)]);
    if (cause !== undefined) {
      this.cause = cause;
    }
  }
}
